walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,85 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Stack;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    ArrayList<TimeRange> meetingOptions = new ArrayList<>();
+
+    //no options for too long of a request
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+        return meetingOptions;
+    }
+
+    //anytime works if no events or no attendees in the request
+    Collection<String> requestAttendees = request.getAttendees();
+    if (requestAttendees.isEmpty() || events.isEmpty()){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    ArrayList<Integer> startTime = new ArrayList<>();
+    ArrayList<Integer> endTime = new ArrayList<>();
+    Collection<String> eventAttendees = new ArrayList<>();
+    for (Event event : events){
+        TimeRange timeRange = event.getWhen();
+        int start = timeRange.start();
+        int end = timeRange.end();
+        startTime.add(start);
+        endTime.add(end);
+        eventAttendees.addAll(event.getAttendees());
+    }
+    
+    //anytime works if event attendees and request attendees are non-overlapping
+    Set<String> attendeesCheckSet = new HashSet<String>();
+    attendeesCheckSet.addAll(requestAttendees);
+    attendeesCheckSet.addAll(eventAttendees);
+    if (attendeesCheckSet.size() == requestAttendees.size()+eventAttendees.size()) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    ArrayList<Integer> allTime = new ArrayList<>();
+    allTime.addAll(startTime);
+    allTime.addAll(endTime);
+    Collections.sort(allTime);
+    
+    //merge all overlapping events
+    ArrayList<Integer> newTimeline = new ArrayList<>();
```

Comment:
Couldn't you make this an array of pairs/time ranges? Then you don't have to worry about iteration by 2 on line 88

Reply:
> Couldn't you make this an array of pairs/time ranges? Then you don't have to worry about iteration by 2 on line 88

Would that be O(n*log(n))?

Reply:
> Couldn't you make this an array of pairs/time ranges? Then you don't have to worry about iteration by 2 on line 88

Yeah I can do that!